crypto/include/internal/evp_int.h

```diff
@@ -369,6 +369,7 @@ struct evp_pkey_st {
     CRYPTO_REF_COUNT references;
     const EVP_PKEY_ASN1_METHOD *ameth;
     ENGINE *engine;
+    ENGINE *pmeth_engine; /* If not NULL public key ENGINE to use */
```

Comment:
I am slightly confused by the existing `engine` member of this structure and how it differs from `pmeth_engine`

Reply:
The existing `engine` pointer is set if the ASN.1 operations are associated with an ENGINE and is then also used for public key operations. An example of this is the GOST ENGINE which performs both ASN.1 and public key operations in an ENGINE. There will be cases where an ENGINE performs public key operations only and ASN.1 operations are handled either by OpenSSL itself or a different ENGINE. That's what `pmeth_engine` is for.